Content.Server/GameTicking/GamePreset.cs

```diff
@@ -18,6 +31,80 @@ public abstract class GamePreset
 
         public virtual void OnGameStarted() { }
 
+        /// <summary>
+        /// Called when a player is spawned in (this includes, but is not limited to, before Start)
+        /// </summary>
+        public virtual void OnSpawnPlayerCompleted(IPlayerSession session, IEntity mob, bool lateJoin) { }
+
+        /// <summary>
+        /// Called when a player attempts to ghost.
+        /// </summary>
+        public virtual void OnGhostAttempt(IConsoleShell? shell, IPlayerSession player, bool canReturnGlobal)
```

Comment:
Kinda weird it takes in the shell. Couldn't you make it a bool and let the shell handle it?

Otherwise seems good.

Reply:
Done.